### PR TITLE
Fix issue 359 - unescape definition in swagger

### LINKED ISSUE #359: Rest Swagger - $ref with `[` and `]` not parsed successfully
When ref with "#/definitions/KeyValuePair[String,String]", the `[` and `]` are decoded to added as key in dict, but it's get without decoding.

Expected behavior: Swagger is parsed successfully.
Actual behavior: Could not resolve reference '#/definitions/KeyValuePair[String,String]' in the document

```json
{
  "definitions": {
    "A": {
      "type": "object",
      "properties": {
        "properties": {
          "description": "Properties",
          "type": "array",
          "items": {
            "$ref": "#/definitions/KeyValuePair[String,String]"
          }
        }
      }
    },
    "KeyValuePair[String,String]": {
      "type": "object",
      "properties": {
        "key": {
          "type": "string",
          "readOnly": true
        },
        "value": {
          "type": "string",
          "readOnly": true
        }
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by hellosnow**
To fix #359 , unescape swagger definition key

@vwxyzh @ansyral @chenkennt @qinezh @superyyrrzz

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        /// <summary>
+        /// Use JSON String Representation instead of URI Fragment Identifier Representation, refer to: https://tools.ietf.org/html/rfc6901#section-5
+        /// Reverse to reference unescape described in http://tools.ietf.org/html/rfc6901#section-4
+        /// </summary>
+        /// <param name="reference"></param>
+        /// <returns></returns>
+        public static string FormatDefinitionSinglePath(string reference)
+        {
+            return Uri.UnescapeDataString(reference.Replace("~", "~0").Replace("/", "~1"));
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs]: still need to careful consideration

**Comment by hellosnow** [comment]
Refer to: https://tools.ietf.org/html/rfc6901#section-5:
- When the json pointer starts with '#/', treat it as **URI Fragment Identifier Representation** and **decode**.
- When the json pointer starts with '/', treat it as **JSON String Representation** and**keep it as**.

@chenkennt @vwxyzh please help to check whether the logic is correct.

File: src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.DocAsCode.Build.RestApi
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs]: add one line before

**Comment by vwxyzh** [comment]
👍

**Pull request merged**